tests/test_models.py

```diff
@@ -475,3 +476,52 @@ def test_sse_event_handler(simai_client, model_factory):
         }
     )
     assert model.is_ready
+
+
+@responses.activate
+def test_post_process_input(simai_client):
+    """WHEN ModelConfiguration includes a specified surface_pp_input arg
+    THEN Model object configuration includes that exact surface_pp_input
+    """
+
+    raw_project = {
+        "id": MODEL_RAW["project_id"],
+        "name": "fifi",
+        "sample": SAMPLE_RAW,
+    }
+
+    responses.add(
+        responses.GET,
+        f"https://test.test/projects/{MODEL_RAW['project_id']}",
+        json=raw_project,
+        status=200,
+    )
+
+    project: Project = simai_client._project_directory._model_from(raw_project)
+    project.verify_gc_formula = Mock()
+
+    MODEL_CONFIG = MODEL_CONF_RAW.copy()
+    MODEL_CONFIG["fields"]["surface_pp_input"] = [
+        {"keys": None, "name": "Temperature_1", "unit": None, "format": "value", "location": "cell"}
+    ]
+
+    MODEL_REQUEST = MODEL_RAW.copy()
+    MODEL_REQUEST["configuration"] = MODEL_CONFIG
+
+    surface_pp_input = PostProcessInput(surface=["Temperature_1"])
+
+    config_with_pp_input = ModelConfiguration(
```

Comment:
since `MODEL_CONFIG` already has a key `surface_pp_input`, this `surface_pp_input` you set here will be overwritten from what it exists in `MODEL_CONFIG`. Are you sure this is what you'd like to test here? 

To me, it would be more preferable to set here the configuration using the individual arguments, as in `test_build_with_new_config` including `surface_pp_input`, for letting the constructor of ModelConfig to consume it, and then compare the result against the actual `surface_pp_input`

I mean:

```py
    model_input = ModelInput(surface=[], boundary_conditions=["Vx"])
    model_output = ModelOutput(
        surface=["Pressure", "WallShearStress_0"], volume=["Velocity_0", "Pressure"]
    )
    global_coefficients = [("max(Pressure)", "maxpress")]
    simulation_volume = {
        "X": {"length": 300.0, "type": "relative_to_min", "value": 15.0},
        "Y": {"length": 80.0, "type": "absolute", "value": -80},
        "Z": {"length": 40.0, "type": "absolute", "value": -20.0},
    }

    config_with_pp_input = ModelConfiguration(
        project=project,
        build_preset="debug",
        continuous=False,
        input=model_input,
        output=model_output,
        global_coefficients=global_coefficients,
        simulation_volume=simulation_volume,
        surface_pp_input=surface_pp_input,
    )

    responses.add(
        responses.POST,
        f"https://test.test/projects/{MODEL_REQUEST['project_id']}/model",
        json=MODEL_REQUEST,
        status=200,
    )

    build_model: Model = simai_client.models.build(config_with_pp_input)
    assert build_model.configuration.surface_pp_input.surface == surface_pp_input.surface
```

Reply:
added more assertions for completeness; `pp_input` remains in mock request json otherwise response would be empty